comp/core/workloadmeta/collectors/internal/process/process_collector.go

```diff
@@ -203,6 +242,198 @@ func (c *collector) detectLanguages(processes []*procutil.Process) []*languagemo
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.sysProbeClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("non-ok status code: url %s, status_code: %d, response: `%s`", req.URL, resp.StatusCode, string(body))
+	}
+
+	err = json.Unmarshal(body, &responseData)
+	if err != nil {
+		return nil, err
+	}
+
+	return &responseData, nil
+}
+
+func (c *collector) handleServiceRetries(pid int32) {
+	tries := c.serviceRetries[pid]
+	tries++
+	if tries < maxPortCheckTries {
+		c.serviceRetries[pid] = tries
+	} else {
+		log.Tracef("[pid: %d] ignoring due to max number of retries", pid)
+		c.ignoredPids.Add(pid)
+		delete(c.serviceRetries, pid)
+	}
+}
+
+// getProcessEntitiesFromServices creates Process entities with service discovery data
+func (c *collector) getProcessEntitiesFromServices(pids []int32, pidsToService map[int32]*model.Service) []*workloadmeta.Process {
+	entities := make([]*workloadmeta.Process, 0, len(pids))
+	now := c.clock.Now()
+
+	for _, pid := range pids {
+		service := pidsToService[pid]
+		if service == nil {
+			c.handleServiceRetries(pid)
+			continue
+		}
+
+		// Update the last heartbeat to current time
+		service.LastHeartbeat = now.Unix()
+
+		entity := &workloadmeta.Process{
+			EntityID: workloadmeta.EntityID{
+				Kind: workloadmeta.KindProcess,
+				ID:   strconv.Itoa(service.PID),
+			},
+			Pid:     int32(service.PID),
+			Service: convertModelServiceToService(service),
+		}
+		entities = append(entities, entity)
+	}
+
+	return entities
+}
+
+// convertModelServiceToService converts model.Service to workloadmeta.Service
+func convertModelServiceToService(modelService *model.Service) *workloadmeta.Service {
+	return &workloadmeta.Service{
+		GeneratedName:              modelService.GeneratedName,
+		GeneratedNameSource:        modelService.GeneratedNameSource,
+		AdditionalGeneratedNames:   modelService.AdditionalGeneratedNames,
+		ContainerServiceName:       modelService.ContainerServiceName,
+		ContainerServiceNameSource: modelService.ContainerServiceNameSource,
+		ContainerTags:              modelService.ContainerTags,
+		TracerMetadata:             modelService.TracerMetadata,
+		DDService:                  modelService.DDService,
+		DDServiceInjected:          modelService.DDServiceInjected,
+		CheckedContainerData:       modelService.CheckedContainerData,
+		Ports:                      modelService.Ports,
+		APMInstrumentation:         modelService.APMInstrumentation,
+		Language:                   modelService.Language,
+		Type:                       modelService.Type,
+		CommandLine:                modelService.CommandLine,
+		StartTimeMilli:             modelService.StartTimeMilli,
+		ContainerID:                modelService.ContainerID,
```

Comment:
Can you remove all the fields which are unused or duplicated with what's already in Process? CheckedContainerData, ContainerID, StartTimeMilli, Language, etc.